src/main/java/org/hillel/homework/service/OrderService.java

```diff
@@ -0,0 +1,102 @@
+package org.hillel.homework.service;
+
+import org.hillel.homework.dto.response.OrderResponse;
+import org.hillel.homework.dto.request.OrderItemRequest;
+import org.hillel.homework.dto.request.OrderRequest;
+import org.hillel.homework.entity.OrderItem;
+import org.hillel.homework.entity.Order;
+import org.hillel.homework.entity.Product;
+import org.hillel.homework.exception.ResourceNotFoundException;
+import org.hillel.homework.mapper.OrderItemMapper;
+import org.hillel.homework.mapper.OrderMapper;
+import org.hillel.homework.repository.OrderItemRepository;
+import org.hillel.homework.repository.OrderRepository;
+import lombok.AllArgsConstructor;
+import org.hillel.homework.repository.ProductRepository;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class OrderService {
+    private OrderRepository orderRepository;
+    private OrderMapper orderMapper;
+    private ProductRepository productRepository;
+    private OrderItemMapper orderItemMapper;
+    private OrderItemRepository orderItemRepository;
+
+    public OrderResponse save(Long id, OrderRequest orderRequest) {
+        Order orderEntity = orderMapper.requestToEntity(orderRequest);
+        if (id != null) {
+            orderEntity.setId(id);
+        }
+        orderEntity.setOrderDate(LocalDate.now());
+        List<OrderItem> orderItems = new ArrayList<>();
+        orderEntity.setOrderItems(new ArrayList<>());
```

Comment:
it looks redundant as you have this in Order 
`private List<OrderItem> orderItems = new ArrayList<>();`